packages/providers/upload-cloudinary/src/index.ts

```diff
@@ -42,7 +42,7 @@ export = {
           config.folder = file.path;
         }
 
-        const uploadStream = cloudinary.uploader.upload_stream(
+        const uploadStream = cloudinary.uploader.upload_chunked_stream(
```

Comment:
`upload_chunked_stream` seems to have its own limitations too, was investigating a bit in the docs and read that it does not set the `resource_type` of the file. 

If the `resource_type` parameter is not set when uploading an asset to Cloudinary, the asset is still uploaded successfully. However, Cloudinary may not apply default optimizations and transformations to the uploaded asset for display via the URL.  For example, if you provide a URL for an image without specifying the `resource_type`, you won't be able to apply image-specific transformations like resizing, cropping, or adding effects to the image.

What do you think if , based on the `file.size` available here, we use one or another. Furthermore, there is the [`upload_large`](https://cloudinary.com/documentation/upload_images#chunked_asset_upload) method, which instead of having a default chunk size of 20MB, it optimizes that value.

Reply:
> it does not set the resource_type of the file.

Where do you read that? I only see that their [docs on chunked upload](https://cloudinary.com/documentation/upload_images#chunked_asset_upload) mention:

> By default, when using the upload_large method, files are uploaded as raw files if the resource_type parameter is not specified. For more details about the resource_type option, see Asset types.

Which is quite the opposite - they automatically set the type to "raw". Even so, this is done within the `upload_large` method, [here](https://github.com/cloudinary/cloudinary_npm/blob/966b0f50ba6229df0d35f19138adaca58820414c/lib/uploader.js#L67). The `upload_chunked_stream` does not seem to wiggle with `resource_type` in any way, just as `upload_stream` which Strapi uses now. I.e. there should be no practical difference between `upload_stream` and `upload_chuncked_stream`. I don't observe any problems or changes using `upload_chuncked_stream` on our production.

> If the resource_type parameter is not set when uploading an asset to Cloudinary, the asset is still uploaded successfully. However, Cloudinary may not apply default optimizations and transformations to the uploaded asset for display via the URL. For example, if you provide a URL for an image without specifying the resource_type, you won't be able to apply image-specific transformations like resizing, cropping, or adding effects to the image.

This shouldn't be a problem as this provider explicitly [sets](https://github.com/strapi/strapi/pull/16621/files#diff-85db252f60d7316518d8b0d589fc89f96908ec5202f9de08c86558de4a6b2fd7R33) `resource_type` to `auto`. This PR doesn't change anything in this regard.


> What do you think if , based on the file.size available here, we use one or another.

In light of the above, is there anything to gain from this increased complexity?

> Furthermore, there is the [upload_large](https://cloudinary.com/documentation/upload_images#chunked_asset_upload) method, which instead of having a default chunk size of 20MB (upload_chunked_stream), it optimizes that value.

`upload_large` cannot take a stream - it only takes a file-system  path to a file or an URL. Though, there's [`upload_large_stream`](https://github.com/cloudinary/cloudinary_npm/blob/966b0f50ba6229df0d35f19138adaca58820414c/lib/uploader.js#L112) but that's only a wrapper around `upload_chunked_stream` to set `resource_type` to `raw` which we don't want - that's why I used `upload_chunked_stream` directly.

Reply:
it seems you are completely right :) I miss read the documentation, I will find time during the week to QA this just in case, but it looks good overall :) ! Thank you